pilot/proxy/envoy/discovery_test.go

```diff
@@ -764,17 +741,6 @@ func TestListenerDiscoverySidecarError2(t *testing.T) {
 	}
 }
 
-func TestListenerDiscoveryIngressError(t *testing.T) {
-	_, _, ds := commonSetup(t)
-	mockDiscovery.HostInstancesError = errors.New("mock HostInstances() error")
-	url := fmt.Sprintf("/v1/listeners/%s/%s", "istio-proxy", mock.Ingress.ServiceNode())
-	response := getDiscoveryResponse(ds, "GET", url, t)
-	if response.StatusCode != http.StatusServiceUnavailable {
-		t.Errorf("unexpected error response from discovery: got %v, want %v",
-			response.StatusCode, http.StatusServiceUnavailable)
-	}
-}
-
```

Comment:
why are you removing these tests?

Reply:
The call to HostInstances is removed for Ingress (see discovery.go in this patch). What I have found was that when the Ingress makes the xDS query to Pilot, it always comes with an empty IP address. And thus the call to HostInstances is not valid and it always return empty result. Due to the removal, these tests become invalid. Also see previous comments about removing that call for Ingress.